torch/ao/quantization/_pt2e/qat_utils.py

```diff
@@ -0,0 +1,137 @@
+import copy
+import operator
+from typing import Any, Callable, Tuple
+
+import torch
+import torch._dynamo
+from torch.fx import GraphModule, Node
+from torch.fx.subgraph_rewriter import _replace_pattern
+import torch.nn.functional as F
+
+
+# Example inputs for both `_conv_bn_pattern` and `_fused_qat_conv_bn_pattern`
+_conv_bn_pattern_example_inputs = (
+    torch.randn(1, 1, 3, 3),  # x
+    torch.randn(1, 1, 1, 1),  # conv_weight
+    torch.randn(1),           # conv_bias
+    torch.randn(1),           # bn_weight
+    torch.randn(1),           # bn_bias
+    torch.randn(1),           # bn_running_mean
+    torch.randn(1),           # bn_running_var
+)
+
+def _conv_bn_pattern(
+    x: torch.Tensor,
+    conv_weight: torch.Tensor,
+    conv_bias: torch.Tensor,
+    bn_weight: torch.Tensor,
+    bn_bias: torch.Tensor,
+    bn_running_mean: torch.Tensor,
+    bn_running_var: torch.Tensor,
+) -> torch.Tensor:
+    x = F.conv2d(x, conv_weight, conv_bias)
+    x = F.batch_norm(x, bn_running_mean, bn_running_var, bn_weight, bn_bias, training=True)
+    return x
+
+def _fused_qat_conv_bn_pattern(
+    x: torch.Tensor,
+    conv_weight: torch.Tensor,
+    conv_bias: torch.Tensor,
+    bn_weight: torch.Tensor,
+    bn_bias: torch.Tensor,
+    bn_running_mean: torch.Tensor,
+    bn_running_var: torch.Tensor,
+) -> torch.Tensor:
+    """
+    Approximated method to fuse conv and bn. It requires only one forward pass.
```

Comment:
Is this copied over from quantizable ConvBN? If so can you reference the code pointer.

Reply:
Yes it's referenced two lines below this